lib/kvdb/ikvdb.c

```diff
@@ -394,15 +424,40 @@ ikvdb_storage_add(const char *kvdb_home, struct kvdb_cparams *params)
         return err;
     }
 
+    err = ikvdb_is_pmem_only(kvdb_home, &meta, &pmem_only);
+    if (err)
+        return err;
+
+    if (!pmem_only && (params->storage.mclass[MP_MED_CAPACITY].path[0] != '\0')) {
+        log_err("Cannot add %s to a standard KVDB %s: capacity mclass path must be provided "
+                "at create time", params->storage.mclass[MP_MED_CAPACITY].path, kvdb_home);
+        return merr(EINVAL);
+    }
+
+    if (pmem_only && (params->storage.mclass[MP_MED_STAGING].path[0] != '\0') &&
+        (params->storage.mclass[MP_MED_CAPACITY].path[0] == '\0')) {
+        log_err("Cannot add %s to a pmem-only KVDB %s: capacity mclass must be added "
+                "before staging", params->storage.mclass[MP_MED_STAGING].path, kvdb_home);
+        return merr(EINVAL);
+    }
```

Comment:
1. Make "at create time" and "before staging" part of the format string in the above calls to `log_err`.
2. Why didn't the printf arg check catch the fact that there are too many args for the format strings?

Reply:
Sure, will combine..

My eyes are unable to catch which log_err has too many args... Can you please help?

Reply:
Ugh... I was fooled by the string concat.  My eyes put a comma at the end of the first line.